src/cloudant/database.py

```diff
@@ -1098,3 +1101,123 @@ def get_query_result(self, selector, fields=None, raw_result=False,
+        :param str group_field: A string field by which to group search matches.
+            Fields containing other data (numbers, objects, arrays)
+            can not be used.
+        :param int group_limit: Maximum group count.
+            This field can only be used if ``group_field`` query parameter
+            is specified.
+        :param group_sort: A JSON field that defines the order of the
+            groups in a search using ``group_field``. The default sort order
+            is relevance. This field can have the same values as the sort field,
+            so single fields as well as arrays of fields are supported.
+        :param int limit: Limit the number of the returned documents to the
+            specified count. In case of a grouped search, this parameter limits
+            the number of documents per group.
+        :param ranges: A JSON facet syntax that reuses the standard Lucene
+            syntax to return counts of results which fit into each specified
+            category. Inclusive range queries are denoted by brackets.
+            Exclusive range queries are denoted by curly brackets.
+            For example ``ranges={"price":{"cheap":"[0 TO 100]"}}`` has an
+            inclusive range of 0 to 100.
+            Requires the index to have faceting enabled.
+        :param sort: A JSON string of the form ``fieldname<type>`` for ascending
+            or ``-fieldname<type>`` for descending sort order. Fieldname is the
+            name of a string or number field and type is either number or string
+            or a JSON array of such strings. The type part is optional and
+            defaults to number.
+        :param str stale: Allow the results from a stale index to be used. This
+            makes the request return immediately, even if the index has not been
+            completely built yet.
+        :param list highlight_fields: A list of fields which should be
+            highlighted.
+        :param str highlight_pre_tag: A string inserted before the highlighted
+            word in the highlights output.  Defaults to ``<em>``.
+        :param str highlight_post_tag: A string inserted after the highlighted
+            word in the highlights output.  Defaults to ``<em>``.
+        :param int highlight_number: A number of fragments returned in
+            highlights. If the search term occurs less often than the number of
+            fragments specified, longer fragments are returned.  Default is 1.
+        :param int highlight_size: A number of characters in each fragment for
+            highlights.  Defaults to 100 characters.
+        :param list include_fields: A list of field names to include in search
+            results. Any fields included must have been indexed with the
+            ``store:true`` option.
+
+        :returns: Search query result data in JSON format
+        """
+        data = dict(query=query)
+        data.update(kwargs)
```

Comment:
Unfortunately, I believe that this creates a vulnerability where someone can provide a `query` as part of the `kwargs` and basically overwrite the `query` provided as a separate argument.  This could cause a lot of confusion for a developer that mistakenly falls into this trap.

I think it would be cleaner/safer to change the method signature to `def get_search_result(self, doc_id, index_name, **query_params)` and to then perform a check for `query` since it is mandatory.  As in:

```python
def get_search_result(self, doc_id, index_name, **query_params):
    ....
    if not query_params.get('query'):
        # raise CloudantArgumentError exception
    data = dict(query_params)
    ....
```